### PR TITLE
feat: Change lambda example to container

### PR DESCRIPTION
**Pull request by candonov**
### What does this PR do?

- switches the sqs-lambda-s3 example to use container image instead of s3 zip file
- updated READMEs

### Motivation

- we don't have examples with container lambda
-  version updates of image tag show on the AWS Console

### More

- [X] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)

- [ ] Yes, I have added a new example under [examples](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/examples) to support my PR

- [X] Yes, I have updated the [docs](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/doc) for this feature

- [ ] Yes, I have linked to an issue or feature request (applicable to PRs that solves a bug or a feature request)

**Note**:
 - Not all the PRs require examples and docs
 - We prefer small, well tested pull requests. Please ensure your pull requests are self-contained, and commits are squashed

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: compositions/upbound-aws-provider/lambda/container.yaml [hunk 1]
```diff
@@ -46,6 +95,10 @@ spec:
           deletionPolicy: Delete
           forProvider:
             packageType: Image
+            role: "tbd"
+            environment:
+              - variables:
+                  hardcoded: value
```
Comment (csantanapr) [inline compositions/upbound-aws-provider/lambda/container.yaml:101]:
```suggestion
            environment:
              - variables: {}
```
can make this shorter like this

File: examples/upbound-aws-provider/composite-resources/serverless-examples/sqs-lambda-s3/README.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ This example deploys the architecture depicted on the diagram. First, it applies
 
 ## Pre-requisites:
  - [Upbound AWS Provider Crossplane Blueprint Examples](../../../README.md)
- - [This serverless application](../object-processor-app/README.md)
+ - [Option 1: Container of this Go application](../object-processor-app/README.md)
```
Comment (csantanapr) [inline examples/upbound-aws-provider/composite-resources/serverless-examples/sqs-lambda-s3/README.md:8]: It mentions "Option 1:" where is option 2 in this doc?

File: examples/upbound-aws-provider/composite-resources/serverless-examples/object-processor-app/README.md [hunk 3]
```diff
@@ -4,12 +4,74 @@ This is a test Go application for serverless examples.
...
+```shell
+cd ../object-processor-app/
 ```
-cd composite-resources/serverless/app/
+## Option 1: Container
+This option contains commands to build the Go binary, build a Docker image, and create ECR repo and upload the image to it.
+First, set the region, ECR_URL, and IMAGE_NAME as environment variables.
+```shell
+export AWS_REGION=<replace-me-with-aws-region> # this should make aws cli pointing to the region without explicitly passing --region 
+export ECR_URL=$(aws sts get-caller-identity|jq -r ".Account" | tr -d '[:space:]').dkr.ecr.$AWS_REGION.amazonaws.com
```
Comment (csantanapr) [inline examples/upbound-aws-provider/composite-resources/serverless-examples/object-processor-app/README.md:18]:
```suggestion
export ECR_URL=$(aws sts get-caller-identity --output json | jq -r ".Account" | tr -d '[:space:]').dkr.ecr.$AWS_REGION.amazonaws.com
```
if not I get error
```
parse error: Invalid numeric literal at line 1, column 8
```

File: examples/upbound-aws-provider/composite-resources/serverless-examples/object-processor-app/README.md [hunk 4]
```diff
@@ -4,12 +4,74 @@ This is a test Go application for serverless examples.
...
+```shell
+GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o main main.go
+```
+Set the `DOCKER_IMAGE` env var, this step is nessasary to properly escape the ':' for the subsequent docker commands
+```shell
+export DOCKER_IMAGE=$ECR_URL/$IMAGE_NAME\:latest
+```
+Build the docker image
+```shell
+docker build -t $DOCKER_IMAGE .
```
Comment (csantanapr) [inline examples/upbound-aws-provider/composite-resources/serverless-examples/object-processor-app/README.md:39]: We should provide `finch` command in addition of `docker`, I don't have "docker" installed on my Mac (I'm not allowed)
Re finch: https://github.com/runfinch/finch

File: examples/upbound-aws-provider/composite-resources/serverless-examples/object-processor-app/README.md [hunk 5]
```diff
@@ -4,12 +4,74 @@ This is a test Go application for serverless examples.
...
+This option contains commands to build the Go binary, build a Docker image, and create ECR repo and upload the image to it.
+First, set the region, ECR_URL, and IMAGE_NAME as environment variables.
+```shell
+export AWS_REGION=<replace-me-with-aws-region> # this should make aws cli pointing to the region without explicitly passing --region 
+export ECR_URL=$(aws sts get-caller-identity|jq -r ".Account" | tr -d '[:space:]').dkr.ecr.$AWS_REGION.amazonaws.com
+export IMAGE_NAME=<replace-me-with-image-name>
 ```
+Log in to ECR
+```shell
+aws ecr get-login-password | docker login --username AWS --password-stdin $ECR_URL
```
Comment (csantanapr) [inline examples/upbound-aws-provider/composite-resources/serverless-examples/object-processor-app/README.md:23]: provide alternative for finch

```
aws ecr get-login-password | finch login --username AWS --password-stdin $ECR_URL
```
I don't have docker for Mac

**Review by csantanapr — APPROVED** [review]
minor suggestions other than that /lgtm

File: examples/upbound-aws-provider/composite-resources/serverless-examples/sqs-lambda-s3/README.md (hunk 2 above)
Comment (candonov) [reply examples/upbound-aws-provider/composite-resources/serverless-examples/sqs-lambda-s3/README.md:8]: It is in the application readme: https://github.com/candonov/crossplane-on-eks/tree/change-lambda-to-container/examples/upbound-aws-provider/composite-resources/serverless-examples/object-processor-app#option-2-zip-file-to-s3

File: examples/upbound-aws-provider/composite-resources/serverless-examples/object-processor-app/README.md (hunk 4 above)
Comment (candonov) [reply examples/upbound-aws-provider/composite-resources/serverless-examples/object-processor-app/README.md:39]: Docker is the de-facto standard, I prefer to keep it to just docker. If you are using flinch or podman, you can extrapolate the relative command. We cannot/should not include every possible option.

**Pull request merged**